packages/eslint-plugin-orbit-components/src/rules/defaultTheme.ts

```diff
@@ -0,0 +1,126 @@
+import * as t from "@babel/types";
+import type { Rule } from "eslint";
+
+export const ERRORS = {
+  variableDeclaration: "Do not use defaultTheme as value, use useTheme() hook or ThemeConsumer",
+  styled:
+    "Do not use defaultTheme directly inside styled components, either create defautlProps or consume the theme from with arrow function expression.",
+  import: "Do not name defaultTheme import as theme, please rename it to defaultTheme/themeDefault",
+  desctructured: "Do not destructure defaultTheme, use useTheme() hook or ThemeConsumer",
+};
+
+export default {
+  meta: {
+    type: "problem",
+    docs: {
+      description: "Rule helps to avoid bad patterns of defaultTheme usage",
+      category: "Possible errors",
+      recommended: true,
+    },
+  },
+
+  create: (context: Rule.RuleContext) => {
+    let specifier = "";
+
+    return {
+      ImportDeclaration(node: t.ImportDeclaration) {
+        if (node.source.value === "@kiwicom/orbit-components/lib/defaultTheme") {
+          node.specifiers.forEach(s => {
+            if (t.isImportDefaultSpecifier(s)) {
+              if (s.local.name === "theme") {
+                context.report({
+                  // @ts-expect-error TODO
+                  node,
+                  message: ERRORS.import,
+                });
```

Comment:
I would personally avoid this part, or at least making it optional, considering that it's a matter of personal preference how you name it 🤔 I can imagine this not being perceived well and resulting in quite a few errors without it pointing to an actual problem, unlike the rest of the rules.

Reply:
I would left it as it is, do not think it will cause any problem, the reason why I did it is that theme imported from `defaultTheme` should be used only inside SC defaultProps and Provider. `theme` is for consumer, for instance. I think we should have consistent names for that import.